packages/ui/src/utils/makeLocalizable.ts

```diff
@@ -220,3 +220,95 @@ export const makeLocalizeable = (resource: LocalizationResource) => {
     translateError,
   };
 };
+
+export type ComplexityErrors = {
+  [key in keyof Partial<Omit<PasswordSettingsData, 'disable_hibp' | 'min_zxcvbn_strength' | 'show_zxcvbn'>>]?: boolean;
+};
+
+export type UsePasswordComplexityConfig = Omit<
+  PasswordSettingsData,
+  'disable_hibp' | 'min_zxcvbn_strength' | 'show_zxcvbn'
+>;
+
+function listFormatSupportedLocalesOf(locale?: string | string[]) {
+  if (!locale) {
+    return false;
+  }
+  const locales = Array.isArray(locale) ? locale : [locale];
+  return (Intl as any).ListFormat.supportedLocalesOf(locales).length === locales.length;
+}
+
+/**
+ * Intl.ListFormat was introduced in 2021
+ * It is recommended to first check for browser support before using it
+ */
+export function canUseListFormat(locale: string | undefined) {
+  return 'ListFormat' in Intl && listFormatSupportedLocalesOf(locale);
+}
+export const addFullStop = (string: string | undefined) => {
+  return !string ? '' : string.endsWith('.') ? string : `${string}.`;
+};
+
+export const createListFormat = (message: string[], locale: string) => {
+  let messageWithPrefix: string;
+  if (canUseListFormat(locale)) {
+    const formatter = new Intl.ListFormat(locale, { style: 'long', type: 'conjunction' });
+    messageWithPrefix = formatter.format(message);
+  } else {
+    messageWithPrefix = message.join(', ');
+  }
+
+  return messageWithPrefix;
+};
+
+const errorMessages: Record<keyof Omit<ComplexityErrors, 'allowed_special_characters'>, [string, string] | string> = {
+  max_length: ['unstable__errors.passwordComplexity.maximumLength', 'length'],
+  min_length: ['unstable__errors.passwordComplexity.minimumLength', 'length'],
+  require_numbers: 'unstable__errors.passwordComplexity.requireNumbers',
+  require_lowercase: 'unstable__errors.passwordComplexity.requireLowercase',
+  require_uppercase: 'unstable__errors.passwordComplexity.requireUppercase',
+  require_special_char: 'unstable__errors.passwordComplexity.requireSpecialCharacter',
+};
```

Comment:
I know this is a draft, so feel free to ignore.
Is is possible to moved this under @clerk/shared ?

Reply:
absolutely! where else would we want to use it? I can easily see the case for the first three functions, but I'm curious where else we'd want to use the `errorMessages` object.

Reply:
I believe this code originated from clerk-js, so the goal would be to eliminate duplication across packages

Reply:
Ah gotcha! Given that the UI package will eventually replace the logic within `clerk-js` I think it's probably fine to leave this duplicated for the moment. That being said we should definitely take a look at what we've pulled in and see if there's a case for it existing in a shared package after the merger.

Reply:
Can we at least add a comment that this was pulled from that code? Just so that it's clear that this was duplicated and, eventually, the other work should be replaced. 🫶

Reply:
Aye aye! Added a note in [ed1f767](https://github.com/clerk/javascript/pull/3812/commits/ed1f767ec7380adb9368f7393f5ff3e7a8e4a352)